teps/README.md

```diff
@@ -70,6 +70,12 @@ The TEP `OWNERS` are the **main** owners of the following projects:
 
 ## Merging TEPs
 
+- When creating a new TEP, pick a number for it, equal to the maximum
+  TEP available in the repo and open PRs. The TEP number shall be set
+  in the TEP file name, TEP title and PR title:
+  - file name in the format `teps/<XXXX>-<tep-title>.md`
+  - title in the format `# TEP-XXXX: <tep-title>`
+  - PR title in the format `TEP-XXXX: <tep-title>`
```

Comment:
does this fix the problem? i think folks pretty much need to pick the number right before merging

OR we get faster at merging things with "proposed" status vs leaving them open (not sure how we continue discussing them in that case? in the merged PR maybe?)

Reply:
So I think how it works in k8s is, at some point, a number is assigned (by one of the approvers) and we go from there. I am fine adding the number as a follow-up too.

Reply:
We can also implement it this way, similar to k8s style. I don't mind either way, as long as we fix the issue for authors of picking a number.